javaparser-core-serialization/src/main/java/com/github/javaparser/serialization/JavaParserJsonSerializer.java

```diff
@@ -0,0 +1,62 @@
+package com.github.javaparser.serialization;
+
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.metamodel.BaseNodeMetaModel;
+import com.github.javaparser.metamodel.JavaParserMetaModel;
+import com.github.javaparser.metamodel.PropertyMetaModel;
+import com.github.javaparser.utils.Log;
+
+import javax.json.stream.JsonGenerator;
+import java.util.EnumSet;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Serializes an AST or a partial AST to JSON.
+ */
+public class JavaParserJsonSerializer {
+    public void serialize(Node node, JsonGenerator generator) {
+        requireNonNull(node);
+        Log.info("Serializing Node to JSON.");
+        serialize(null, node, generator);
+    }
+
+    private void serialize(String nodeName, Node node, JsonGenerator generator) {
+        requireNonNull(node);
+        BaseNodeMetaModel nodeMetaModel = JavaParserMetaModel.getNodeMetaModel(node.getClass()).orElseThrow(() -> new IllegalStateException("Unknown Node: " + node.getClass()));
+
+        if (nodeName == null) {
+            generator.writeStartObject();
+        } else {
+            generator.writeStartObject(nodeName);
+        }
+        generator.write("!", node.getClass().getName());
```

Comment:
Perhaps we could use a constant for that or maybe give a way to set this name in some sort of configuration